gnomad/utils/constraint.py

```diff
@@ -360,3 +412,285 @@ def collapse_strand(
+
+    .. note::
+
+        This function expects that the input Table(`coverage_ht`) was created using
+        `get_proportion_observed_by_coverage`, which means that `coverage_ht` should
+        contain only high quality synonymous variants below 0.1% frequency.
+
+        This function also expects that the following fields are present in
+        `coverage_ht`:
+        - context - trinucleotide genomic context
+        - ref - the reference allele
+        - alt - the alternate allele
+        - methylation_level - methylation level
+        - cpg - whether the site is CpG site
+        - exome_coverage - median exome coverage at integer values between 1-100
+        - observed_variants - the number of observed variants in the dataset for each
+        variant. Note that the term "variant" here refers to a specific substitution,
+        context, methylation level, and coverage combination
+        - downsampling_counts_{pop} (optional) - array of observed variant counts per
+        population after downsampling. Used only when `pops` is specified.
+        - mu_snp - mutation rate
+        - possible_variants - the number of possible variants in the dataset for each
+        variant
+
+    :param coverage_ht: Input coverage Table.
+    :param weighted: Whether to weight the plateau models (a linear regression
+        model) by 'possible_variants'. Default is False.
+    :param pops: List of populations used to build plateau models.
+        Default is ().
+    :param keys: Annotations used to group observed and possible variant counts.
+        Default is ("context", "ref", "alt", "methylation_level", "mu_snp").
+    :param cov_cutoff: Median coverage cutoff. Sites with coverage above this cutoff
+        are considered well covered. Default is `COVERAGE_CUTOFF`.
+    :return: Coverage model and plateau models.
+    """
+    # Filter to sites with coverage above `cov_cutoff`.
+    high_cov_ht = coverage_ht.filter(coverage_ht.exome_coverage >= cov_cutoff)
+    agg_expr = {
+        "observed_variants": hl.agg.sum(high_cov_ht.observed_variants),
+        "possible_variants": hl.agg.sum(high_cov_ht.possible_variants),
+    }
+    for pop in pops:
+        agg_expr[f"observed_{pop}"] = hl.agg.array_sum(
+            high_cov_ht[f"downsampling_counts_{pop}"]
+        )
+
+    # Generate a Table with all necessary annotations (x and y listed above)
+    # for the plateau models.
+    high_cov_group_ht = high_cov_ht.group_by(*keys).aggregate(**agg_expr)
+    high_cov_group_ht = annotate_mutation_type(high_cov_group_ht)
+
+    # Build plateau models.
+    plateau_models_agg_expr = build_plateau_models(
+        cpg_expr=high_cov_group_ht.cpg,
+        mu_snp_expr=high_cov_group_ht.mu_snp,
+        observed_variants_expr=high_cov_group_ht.observed_variants,
+        possible_variants_expr=high_cov_group_ht.possible_variants,
+        pop_observed_variants_array_expr=[
+            high_cov_group_ht[f"observed_{pop}"] for pop in pops
+        ],
```

Comment:
it just occurred to me that this line will not work with the function as written (pops defaults to ()). how about something like this
```suggestion
        pop_observed_variants_array_expr=[
            high_cov_group_ht[f"observed_{pop}"] for pop in pops
        ] if pops else None,
```

Reply:
Sorry it's my bad. It worked before I got rid of the redundancy and I forgot to double check. I will fix it!